pennylane/templates/layers/cnot_ring.py

```diff
@@ -0,0 +1,161 @@
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``CnotRingLayers`` template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+from pennylane.ops import CNOT, RX
+from pennylane.templates import broadcast
+from pennylane.templates.utils import (
+    _check_shape,
+    _check_no_variable,
+    _check_wires,
+    _check_number_of_layers,
+    _get_shape,
+)
+
+
+@template
+def CnotRingLayers(weights, wires, rotation=None):
+    r"""Layers consisting of one-parameter single-qubit rotations on each qubit, followed by a closed chain
+    or *ring* of CNOT gates.
+
+    The ring of CNOT gates connects every qubit with its neighbour,
+    whereas the last qubit is considered to be a neighbour of the first qubit.
+
+    .. figure:: ../../_static/templates/layers/cnot_ring.png
+        :align: center
+        :width: 40%
+        :target: javascript:void(0);
+
+    The argument ``weights`` is an array of weights for each layer. The number of layers :math:`L` is derived
+    from the first dimension of ``weights``. When using a single wire, the template only applies the single
+    qubit gates in each layer.
+
+    .. note::
+
+        This template follows the convention of dropping the entanglement between the last and the first
+        qubit when using only two wires, so the entangler is not repeated on the same wires.
+        In this case, only one CNOT gate is applied in each layer:
+
+        .. figure:: ../../_static/templates/layers/cnot_ring_2wires.png
+            :align: center
+            :width: 30%
+            :target: javascript:void(0);
+
+    Args:
+
+        weights (array[float]): array of weights of shape ``(L, len(wires))``
+        wires (Sequence[int] or int): qubit indices that the template acts on
+        rotation (pennylane.ops.Operation): one-parameter single-qubit gate to use,
+                                            if ``None``, :class:`~pennylane.ops.RX` is used as default
+    Raises:
+        ValueError: if inputs do not have the correct format
+
+    .. UsageDetails::
+
+        The template is used inside a qnode:
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import CnotRingLayers
+            from math import pi
+
+            n_wires = 3
+            dev = qml.device('default.qubit', wires=n_wires)
+
+            @qml.qnode(dev)
+            def circuit(weights):
+                CnotRingLayers(weights=weights, wires=range(n_wires))
+                return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_wires)]
+
+        >>> circuit([[pi, pi, pi]])
```

Comment:
Would it make things unintuitive if in this case (`n_layers=1`), we could also accept a list instead of a nested list?

This would require a change in `_check_shape` from what I see (I can imagine users trying - myself included - passing `[pi, pi, pi]` here for the first time):
```python
/pennylane/pennylane/templates/utils.py in _check_shape(inpt, target_shape, msg, bound)
    121     else:
    122         if shape != target_shape:
--> 123             raise ValueError(msg)
    124 
    125     return shape

ValueError: 'weights' must be of shape (3, 3); got (3,)
```

Reply:
Worth thinking about, but I find it more intuitive with the weight shape remaining consistent

Reply:
Me too, we may risk ambiguities.